src/tests/multihost/ad/test_dyndns.py

```diff
@@ -0,0 +1,537 @@
+    new_hostname = "client" + str(random.randrange(99)) + "." + ad_domain
+    session_multihost.client[0].run_command(f'hostnamectl set-hostname \"{new_hostname}\"')
+
+    def restore_hostname():
+        """ Restore hostname """
+        session_multihost.client[0].run_command(f'hostnamectl set-hostname {old_hostname}')
+
+    request.addfinalizer(restore_hostname)
+
+
+@pytest.fixture(scope="class")
+def disable_dns_forwarders(session_multihost, request):
+    """ Disables recursive lookups on DNS server """
+    session_multihost.ad[0].run_command('dnscmd.exe /config /norecursion 1')
+
+    def enable_dns_forwarders():
+        """ Enables recursion lookups on DNS servers """
+        session_multihost.ad[0].run_command('dnscmd.exe /config /norecursion 0')
+
+    request.addfinalizer(enable_dns_forwarders)
+
+
+@pytest.fixture(scope="function")
+def extra_network(session_multihost, request):
+    """ Create reverse zone """
+    dns = ADDNS(session_multihost.ad[0])
+    network = "1.168.192.in-addr.arpa"
+    assert dns.add_zone(network)
+
+    def remove_extra_network():
+        """ Delete reverse zone"""
+        dns.del_zone(network)
+
+    request.addfinalizer(remove_extra_network)
+
+
+@pytest.fixture(scope="function")
+def extra_interface(session_multihost, request):
+    """ Create extra interface """
+    extra_interface = 'ibm' + str(random.randrange(99))
+    extra_ip = '192.168.1.' + str(random.randrange(2, 255))
+    session_multihost.client[0].run_command('ip link add ' + extra_interface + ' type dummy')
+    session_multihost.client[0].run_command('ip addr add ' + extra_ip + ' dev ' + extra_interface)
+
+    def remove_interface():
+        """ Delete interface """
+        session_multihost.client[0].run_command('ip addr flush dev ' + extra_interface)
+        session_multihost.client[0].run_command('ip link del ' + extra_interface)
+
+    request.addfinalizer(remove_interface)
+    return extra_interface, extra_ip
+
+
+@pytest.mark.usefixtures("disable_dns_forwarders", "change_client_hostname")
+@pytest.mark.dyndns
+@pytest.mark.tier1_3
+class TestDynDns(object):
+
+    @staticmethod
+    def test_0001_verify_with_default_setting(multihost, adjoin):
+        """
+        :title: IDM-SSSD-TC: ad_provider: dyndns: verify with default settings
+        :id: 183cc040-0b8c-47af-8b19-8bcb72c3f001
+        :steps:
+          1. Join client to AD domain
+          2. Perform DNS lookup for hostname
+          3. Perform DNS lookup for IP
+        :expectedresults:
+          1. Client joins domain
+          2. Hostname resolves correctly
+          3. IP resolves correctly
+        """
+        adjoin(membersw='adcli')
+        client = sssdTools(multihost.client[0], multihost.ad[0])
+        dns = ADDNS(multihost.ad[0])
+        hostname = multihost.client[0].run_command('hostname').stdout_text.rstrip()
+        ip = multihost.client[0].ip
+
+        client.clear_sssd_cache()
+        assert dns.find_a(hostname)
```

Comment:
The asserts could use a messages stating what exactly failed to make it easier when someone will be looking on failure in the future.
```
assert dns.find_a(hostname), f"Could not resolve a hostname: {hostname}."
```

Reply:
I'll add a comment for when IPV6 support is added, to do this. It will get more complicated with IPV6 and it is a good idea.